### PR TITLE
Add ability to see last run on map at nearly any time*

### LINKED ISSUE #1611: Seeing other company's routes
A general request from our group would be to update the game map for all players with the routes a company has selected.  Currently, only the acting director/president gets to see the routes selected.  Other players would like to see this too, once the director has submitted the routes.

### PR DESCRIPTION
**Pull request by jenf**
* not during run routes/dividends

fixes #1611 & #563

### CODE REVIEW

**Comment by benjaminxscott** [comment]
Would `Show Last Routes For:` be a good descriptive label? 

Then it's more clear the operation that button refers to 
- "Show", similar to the other toggles
- "Last Route(s)", to make it clear this shows all routes
- "For", to make clear that this selector is per-company (and being a bit more clear in the "None" case)

**Comment by benjaminxscott** [comment]
Thanks for addressing out my suggestion - to clarify, I meant it as a pure suggestion rather than a 'review', fwiw

File: assets/app/view/game/map_controls.rb [hunk 1]
```diff
@@ -55,6 +58,69 @@ def starting_map_controls
...
 
+      def generate_last_route(entity)
+        trains = entity.trains
+        operating = entity.operating_history
+        last_run = operating[operating.keys.max]&.routes
+        return [] unless last_run
+
+        halts = operating[operating.keys.max]&.halts
+        routes = []
+        last_run.map do |train, connections|
```
Comment (tobymao) [inline assets/app/view/game/map_controls.rb:69]: why is this a map compact if you don't use the output?

**Comment by roseundy** [comment]
I suspect this might not play nice with the 1860 leased trains - not that you should worry about that.

File: assets/app/view/game/map_controls.rb [hunk 2]
```diff
@@ -55,6 +58,69 @@ def starting_map_controls
...
+          next unless trains.include?(train)
```
Comment (roseundy) [inline assets/app/view/game/map_controls.rb:70]: Do you really want this? Doesn't this mean that a route won't be shown if the corporation no longer owns the train with which the route was run?

**Pull request merged**